### PR TITLE
Health check feature

### PR DESCRIPTION
**Pull request by skp96**
[Link to story](https://trello.com/c/K5N6Kdvj/87-health-check-feature-5)

Highlights:

- Created `GetHtmlHealthCheckAction` which receives a `FileIo`. `FileIo` can read from a resource
- Changed the directory `StructuredDataGenerators` to `Utilities` since some `Actions` depend on support classes 
- Added tests for `GetHtmlHealthCheckAction`, and `FileIo`. Additionally wrote tests to ensure functionality of the health-check feature (e.g `RouterTest`, `RequestParserTest`, and `HttpResponseBuilderTest`

### CODE REVIEW

File: src/test/kotlin/Utilities/FileIoTest.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import Utilities.FileIo
+import kotlin.test.*
+
+class FileIoTest {
```
Comment (k3ntako) [inline src/test/kotlin/Utilities/FileIoTest.kt:4]: might not hurt to test some edge cases:
- empty file
- no file (can be that it just throws an error)

**Review by k3ntako — COMMENTED** [review]
Good work! One small comment

File: src/test/kotlin/Utilities/FileIoTest.kt (hunk 1 above)
Comment (cdemyanovich) [reply src/test/kotlin/Utilities/FileIoTest.kt:4]: Seconded.

File: src/test/kotlin/Utilities/FileIoTest.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+                "    <title>TODO List</title>\n" +
+                "</head>\n" +
+                "\n" +
+                "<body>\n" +
+                "<main>\n" +
+                "    <strong>Status:</strong> pass\n" +
+                "</main>\n" +
+                "</body>\n" +
+                "</html>\n"
+        assertEquals(expectedResult, fileIo.readResource("test-html.html"))
```
Comment (cdemyanovich) [inline src/test/kotlin/Utilities/FileIoTest.kt]: This file doesn't have to be an HTML file. It could be any file, and the content could be very simple, e.g., just one line with one word or two lines with one word each. Such a test file would reduce the string concatenation you have to set up in `expectedResult`.

**Review by cdemyanovich — APPROVED** [review]
